File: services/frontend/src/lib/api/applicants/types/applicant-by-school.ts
export interface ApplicantsBySchool {
  count: number;
  maxCount: number;
  applicants: ApplicantBySchool[];
}

export interface ApplicantBySchool {
  id: string;
  firstname: string;
  lastname: string;
  classLevel: string;
  statusIdentifier: string;
  degree: ApplicantBySchoolDegree;
  address: ApplicantBySchoolAddress
}

export interface ApplicantBySchoolDegree {
  id: string;
  name: string;
}

export interface ApplicantBySchoolAddress {
  zipCode: string;
  city: string;
  country: string;
}

File: services/backend/src/resources/applicants/service/models/applicant_by_school.go
package models

import "github.com/google/uuid"

type ApplicantBySchoolModel struct {
	ID        uuid.UUID
	Firstname string
	Lastname  string

	ClassLevel       string
	StatusIdentifier string
	Degree           *ApplicantBySchoolDegreeModel
	Address          *ApplicantBySchoolAddressModel
}

type ApplicantBySchoolAddressModel struct {
	ZipCode string
	City    string
	Country string
}

type ApplicantBySchoolDegreeModel struct {
	ID   uuid.UUID
	Name string
}
